pkg/destroy/aws/aws.go

```diff
@@ -56,6 +56,11 @@ type ClusterUninstaller struct {
 	Logger    logrus.FieldLogger
 	Region    string
 	ClusterID string
+
+	// Session is the AWS session to be used for deletion.  If nil, a
```

Comment:
what's the reasoning behind defaulting to getting the session from env if not provided?

IMO the destroy code expects the session and uses it. configuration error if it is not setup and error out.

Reply:
> IMO the destroy code expects the session and uses it. configuration error if it is not setup and error out.

This will break our existing consumers, won't it?  I guess they can bump their surrounding code when they bump their vendored copy, but it seemed easy enough to continue to support them with the same behavior we had previously.

Reply:
:+1: